### DISCUSSION TITLE
What can be the maximum byte size for zip_fread() ?

### DISCUSSION THREAD

**Discussion by Purushoth24051999** [nih-at/libzip#472] [General]
I use below function to read what can be the optimal maximum values for `nbytes` parameter

```
zip_int64_t zip_fread(zip_file_t *file, void *buf, zip_uint64_t nbytes);
```

Right now I have passed 64 kb as its value

```
auto length = zip_fread(filepath, buffer, 65536);`
```

**Comment by dillof** [comment]
The maximum that libzip supports is INT64_MAX (9223372036854775807), which is more than your system has available memory.

The optimum depends on your application. Larger values have less overhead since you're calling `zip_fread()` less often but use more memory and may not fit into caches and thus lead to slower memory accesses. 

64k seems like a good compromise.
